packages/next/src/build/get-babel-config-file.ts

```diff
@@ -0,0 +1,30 @@
+import { fileExists } from '../lib/file-exists'
+import { join } from 'path'
+
+const BABEL_CONFIG_FILES = [
+  '.babelrc',
+  '.babelrc.json',
+  '.babelrc.js',
+  '.babelrc.mjs',
+  '.babelrc.cjs',
+  'babel.config.js',
+  'babel.config.json',
+  'babel.config.mjs',
+  'babel.config.cjs',
+]
+
+export async function getBabelConfigFile(
+  dir: string
+): Promise<string | undefined> {
+  const babelConfigFile = await BABEL_CONFIG_FILES.reduce(
+    async (memo: Promise<string | undefined>, filename) => {
+      const configFilePath = join(dir, filename)
+      return (
+        (await memo) ||
+        ((await fileExists(configFilePath)) ? configFilePath : undefined)
+      )
+    },
+    Promise.resolve(undefined)
+  )
+  return babelConfigFile
```

Comment:
I know this is just renaming this file, but this function is _really_ hard to understand. Could we update this to instead be:
```suggestion
  for (const filename of BABEL_CONFIG_FILES) {
    const configFilePath = join(dir, filename)
    const exists = await fileExists(configFilePath)
    if (!exists) continue

    return configFilePath
  }
```

Reply:
Fixed in #54741 🙌